src/main/java/io/kestra/plugin/graphql/Request.java

```diff
@@ -0,0 +1,293 @@
+    )
+    private Property<String> operationName;
+
+    @Builder.Default
+    protected Property<String> method = Property.of("POST");
+
+    @Builder.Default
+    @Schema(
+        title = "Fail task on GraphQL errors",
+        description = "If true, the task will fail when GraphQL returns errors in the response."
+    )
+    private Property<Boolean> failOnGraphQLErrors = Property.of(false);
+
+    @Override
+    protected HttpRequest request(RunContext runContext) throws IllegalVariableEvaluationException, URISyntaxException, IOException {
+
+        String renderedUri = runContext.render(this.uri).as(String.class).map(s -> s.replace(" ", "%20")).orElseThrow();
+        String methodName = runContext.render(this.method).as(String.class).orElse("POST");
+
+        HttpRequest.HttpRequestBuilder requestBuilder = HttpRequest.builder()
+            .method(methodName)
+            .uri(URI.create(renderedUri));
+
+        String renderedQuery = runContext.render(this.query).as(String.class).orElseThrow();
+
+        Map<String, Object> requestPayload = new HashMap<>();
+        requestPayload.put("query", renderedQuery);
+
+        if (variables != null) {
+            Object renderedVariables = runContext.render(variables).asMap(String.class, Object.class);
+            requestPayload.put("variables", renderedVariables);
+        }
+
+        if (operationName != null) {
+            String renderedOpName = runContext.render(operationName).as(String.class).orElse(null);
+            if (renderedOpName != null && !renderedOpName.isEmpty()) {
+                requestPayload.put("operationName", renderedOpName);
+            }
+        }
+
+        requestBuilder.body(HttpRequest.JsonRequestBody.builder()
+            .content(requestPayload)
+            .charset(StandardCharsets.UTF_8)
+            .build());
+
+        var renderedHeader = runContext.render(this.headers).asMap(CharSequence.class, CharSequence.class);
+        if (!renderedHeader.isEmpty()) {
+            requestBuilder.headers(HttpHeaders.of(
+                renderedHeader
+                    .entrySet()
+                    .stream()
+                    .map(throwFunction(e -> new AbstractMap.SimpleEntry<>(
+                            e.getKey().toString(),
+                            runContext.render(e.getValue().toString())
+                        ))
+                    )
+                    .collect(Collectors.groupingBy(AbstractMap.SimpleEntry::getKey, Collectors.mapping(AbstractMap.SimpleEntry::getValue, Collectors.toList()))),
+                (a, b) -> true)
+            );
+        }
+
+        return requestBuilder.build();
+    }
+
+    @Override
+    public Output run(RunContext runContext) throws Exception {
+
+        try (HttpClient client = this.client(runContext)) {
+            HttpRequest request = request(runContext);
+
+            HttpResponse<Byte[]> response = client.request(request, Byte[].class);
```

Comment:
```suggestion
            HttpResponse<byte[]> response = client.request(request, byte[].class);
```

You should be able to directly use a byte array